src/Xamarin.Android.Build.Tasks/Tasks/GenerateLayoutBindings.CSharpBindingGenerator.cs

```diff
@@ -239,7 +240,7 @@ protected override void WriteLocationDirective (State state, LayoutWidget widget
 				if (loc == null)
 					return;
 
-				WriteLineIndent (state, $"#line {loc.Line} \"{loc.FilePath}\"");
+				WriteLineIndent (state, $"#line {loc.Line} \"{Path.GetFullPath (loc.FilePath)}\"");
```

Comment:
The code from xaprepare was taken from the BCL, almost verbatim, precisely because it wasn't available to us. I think it's the best way to work around this limitation.

Reply:
doh :( ... I might just move it to its own self contained class then... so we can share the code.